### PR TITLE
Add CONTRIBUTE.md

### LINKED ISSUE #18: Write a format style guide

### PR DESCRIPTION
**Pull request by Bl4ckb0ne**
Fixes #18

### CODE REVIEW

File: CONTRIBUTE.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
+# Contributing to bspwc
+
+Contributing only involves opening a pull request or an issues. You'll be more
```
Comment (wizbright) [inline CONTRIBUTE.md]: Issues should be issue as it is a singular item.

File: CONTRIBUTE.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+1. git fetch upstream
+2. git checkout -b add-so-and-so-feature upstream/master
+3. work
+4. git push -u origin add-so-and-so-feature
+5. Make pull request from your feature branch
+
+## Commit Messages
+
+Please strive to write good commit messages. Here's some guidelines to follow:
```
Comment (wizbright) [inline CONTRIBUTE.md]: informative instead of good as good is relative.

File: CONTRIBUTE.md [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+bspwc is written in C. The style guidelines is [kernel
+style](https://www.kernel.org/doc/Documentation/process/coding-style.rst).
+Some points to note:
+
+* Do not use typedefs unless you have a good reason
+* Do not use macros unless you have a *really* good reason
+* Tabs, not spaces
+* `char* pointer` - note position of `*`
+* Use logging with reckless abandon
+* Braces goes on a new line
```
Comment (wizbright) [inline CONTRIBUTE.md]: braces "go" on a new line

File: CONTRIBUTE.md [hunk 4]
```diff
@@ -76,7 +77,8 @@ Some points to note:
 * Tabs, not spaces
 * `char* pointer` - note position of `*`
 * Use logging with reckless abandon
-* Braces goes on a new line
+* Braces go on a new line
+* One tab for case within a switch
```
Comment (wizbright) [inline CONTRIBUTE.md]: One tab per case within a switch statement

**Comment by wizbright** [comment]
Looks like #19 is done then.

**Comment by Bl4ckb0ne** [comment]
Almost, I'll ping you when I'll add more stuff

**Pull request merged**